java/java.mx.project/src/org/netbeans/modules/java/mx/project/SuiteActionProvider.java

```diff
@@ -0,0 +1,245 @@
+                        if (segments.length > 2) {
+                            File src = new File(segments[0]);
+                            if (src.exists()) {
+                                int lineNumber = parseLineNumber(segments) - 1;
+                                return Collections.singletonList(ConvertedLine.forText(line, new OutputListener() {
+                                    @Override
+                                    public void outputLineSelected(OutputEvent ev) {
+                                        openLine(Line.ShowOpenType.NONE, Line.ShowVisibilityType.FRONT);
+                                    }
+
+                                    @Override
+                                    public void outputLineAction(OutputEvent ev) {
+                                        openLine(Line.ShowOpenType.OPEN, Line.ShowVisibilityType.FOCUS);
+                                    }
+
+                                    private boolean openLine(final Line.ShowOpenType openType, final Line.ShowVisibilityType visibilityType) throws IndexOutOfBoundsException {
+                                        FileObject fo = FileUtil.toFileObject(src);
+                                        if (fo != null) {
+                                            Lookup lkp = fo.getLookup();
+                                            final LineCookie lines = lkp.lookup(LineCookie.class);
+                                            if (lines != null) {
+                                                Line open = lines.getLineSet().getOriginal(lineNumber);
+                                                if (open != null) {
+                                                    open.show(openType, visibilityType);
+                                                    return true;
+                                                }
+                                            }
+                                            Openable open = lkp.lookup(Openable.class);
+                                            if (open != null) {
+                                                open.open();
+                                            } else {
+                                                Toolkit.getDefaultToolkit().beep();
+                                            }
+                                        }
+                                        return false;
+                                    }
+
+                                    @Override
+                                    public void outputLineCleared(OutputEvent ev) {
+                                    }
+                                }));
+                            }
+                        }
+                        return null;
+                    };
+                });
+        ProcessBuilder processBuilder = ProcessBuilder.getLocal();
+        processBuilder.setWorkingDirectory(suiteDir.getPath());
+        processBuilder.setExecutable("mx"); // NOI18N
```

Comment:
This seems to assume `mx` is on the IDE process'  `PATH`. Shouldn't it be configurable in Tools | Options, similar to maven/ant, especially when NB does not distribute `mx` ? If reasonable, I recommend to track it as JIRA enhancement.

Reply:
True, the UI should be improved in the future. Recorded as [NETBEANS-5158](https://issues.apache.org/jira/browse/NETBEANS-5158).